File: core/src/main/java/com/xmartlabs/base/core/module/AndroidModule.java
package com.xmartlabs.base.core.module;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.xmartlabs.base.core.model.BuildInformation;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AndroidModule {
  private final Application application;

  public AndroidModule(Application application) {
    this.application = application;
  }

  @Provides
  @Singleton
  public Context provideApplicationContext() {
    return application;
  }

  @Provides
  @Singleton
  public BuildInformation provideBuildInformation() {
    return null;
  }

  @Provides
  @Singleton
  public SharedPreferences provideSharedPreferences() {
    return PreferenceManager.getDefaultSharedPreferences(application);
  }
}

File: core/src/main/java/com/xmartlabs/bigbang/core/module/AndroidModule.kt
package com.xmartlabs.bigbang.core.module

import android.app.Application
import android.content.Context
import android.preference.PreferenceManager
import com.xmartlabs.bigbang.core.model.BuildInfo
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AndroidModule {
  @Provides
  @Singleton
  fun provideApplicationContext(application: Application): Context = application

  @Provides
  @Singleton
  fun provideSharedPreferences(context: Context) = PreferenceManager.getDefaultSharedPreferences(context)
}
